packages/monaco/src/browser/monaco-keybinding.ts

```diff
@@ -50,7 +50,7 @@ export class MonacoKeybindingContribution implements KeybindingContribution {
                             keybinding = { ...keybinding, ctrlKey: true, metaKey: false };
                         }
                     }
-                    const isInDiffEditor = item.when && /(?<!\!\s*)isInDiffEditor/gm.test(item.when.serialize());
+                    const isInDiffEditor = item.when && /([^!]|\s*)isInDiffEditor/gm.test(item.when.serialize());
```

Comment:
Ah, because I reused the same regex object for my tests, turns out it wronged the results.

TODO: Find an other regex.